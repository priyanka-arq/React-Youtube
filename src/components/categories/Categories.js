import React from "react";
import "./_categories.scss";
import { useState } from "react";
import { HiChevronLeft, HiChevronRight } from "react-icons/hi";
import { useDispatch } from "react-redux";
import {
  getPopularVideos,
  getVideosByCategory,
} from "../../redux/actions/videosAction";
export default function Categories() {
  const [activeElement, setActiveElement] = useState("All");
  const dispatch = useDispatch();
  const keywords = [
    "All",
    "Javascript",
    "React js",
    "Angular js",
    "React Native",
    "use of API",
    "Redux",
    "Poor Coder",
    "Algorithm Art ",
    "Meditation music",
    "Music",
    "Thrillers",
    "Courses",
    "Indian soap operas",
    "ios",
    "Conversation",
    "Comedy",
    "Live",
    "Nature",
    "T-Series",
    "Music",
    "Guitar",
    "Bengali Songs",
    "Coding",
    "Cricket",
    "Football",
  ];

  const handleClick = (value) => {
    if (value === "All") {
      dispatch(getPopularVideos());
    } else {
      dispatch(getVideosByCategory(value));
      // getVideosByCategory(value, dispatch);
    }

    setActiveElement(value);
  };
  return (
    <div className="categories" data-test="categoriesComponent">
      <HiChevronLeft />
      {keywords.map((keyword, i) => (
        <span
          data-test="buttonComponent"
          className={activeElement === keyword ? "active" : ""}
          key={i}
          onClick={() => handleClick(keyword)}
        >
          {keyword}
        </span>
      ))}
      <HiChevronRight />
    </div>
  );
}
