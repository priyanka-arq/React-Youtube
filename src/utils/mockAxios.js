//############################//

//inside src => craete __mocks__ => axios.js
//paste following code

const mockAxios = jest.genMockFromModule("axios");

// this is the key to fix the axios.create() undefined error!
mockAxios.create = jest.fn(() => mockAxios);
mockAxios.post = jest.fn(() => mockAxios);
mockAxios.get = jest.fn(() => mockAxios);

export default mockAxios;

// export default {
//   get: jest.fn(() => Promise.resolve({ data: {} })),
//   create: jest.fn(() => Promise.resolve({ data: {} })),
// };

//#########################################

import mockAxios from "axios";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

import {
  getCommentsDetails,
  addComment,
} from "../redux/actions/commentListAction";

import {
  COMMENTS_LIST_REQUEST,
  COMMENTS_LIST_SUCCESS,
  CREATE_COMMENT_SUCCESS,
  CREATE_COMMENT_FAILURE,
  COMMENTS_LIST_FAILURE,
} from "../redux/actionType";

let actions;
let store;
const createMockStore = configureMockStore([thunk]);

describe("get comments list action", () => {
  it("it returns data when api is called", async function () {
    const data = { response: true };
    const comments = [
      {
        kind: "youtube#commentThread",
        etag: "9SadOUrGkPQSt0sorsFijKTBL80",
        id: "Ugwybolefx_P4E6phah4AaABAg",
        snippet: {},
      },
      {
        kind: "youtube#commentThread",
        etag: "9SadOUrGkPQSt0sorsFijKTBL80",
        id: "Ugwybolefx_P4E6phah4AaABAg",
        snippet: {},
      },
    ];
    //mock.onGet("/commentThreads").reply(200, comments);

    mockAxios.get.mockImplementationOnce(() =>
      Promise.resolve({
        data: { results: comments },
      })
    );

    // expect(mockAxios.get).toHaveBeenCalledTimes(1);

    store = createMockStore({
      commentList: [
        {
          kind: "youtube#commentThread",
          etag: "9SadOUrGkPQSt0sorsFijKTBL80",
          id: "Ugwybolefx_P4E6phah4AaABAg",
          snippet: {},
        },
        {
          kind: "youtube#commentThread",
          etag: "9SadOUrGkPQSt0sorsFijKTBL80",
          id: "Ugwybolefx_P4E6phah4AaABAg",
          snippet: {},
        },
      ],
    });

    store.dispatch(getCommentsDetails("abc")).then(() => {
      expect(store.getActions()).toContainEqual({
        type: COMMENTS_LIST_REQUEST,
      });
    });

    //expect(mock.history.get.length).toBe(1);
  });
});

///###########################
describe("add comment", () => {
  const store = createMockStore({
    auth: {
      accessToken: "AIzaSyCLg4cHF4d0Sj_UwQQhV1JPvCngteYtNkM",
    },
  });

  test("add comment action", () => {
    const comment = "nice";
    const auth = { accessToken: "AIzaSyCLg4cHF4d0Sj_UwQQhV1JPvCngteYtNkM" };

    mockAxios.post("/commentThreads", comment, {
      params: {
        part: "snippet ",
      },
      headers: {
        Authorization: `Bearer ${auth.accessToken}`,
      },
    });

    expect(mockAxios.post).toHaveBeenCalledTimes(1);

    store.dispatch(addComment("abc", "xyz")).then(() => {
      actions = store.getActions();
      expect(store.getActions()).toMatchSnapshot();

      expect(actions[0]).toEqual({
        type: CREATE_COMMENT_SUCCESS,
      });
    });
  });
});

describe("commentList action", () => {
  const store = createMockStore({});

  it("gets a comments list", () =>
    store.dispatch(getCommentsDetails("abc")).then(() =>
      expect(store.getActions()).toContainEqual({
        type: COMMENTS_LIST_FAILURE,
      })
    ));
});

describe("fetches comments ", async () => {
  const comments = [
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
  ];

  it("should execute fetch data", () => {
    store = createMockStore();
    // Return the promise
    return store.dispatch(getCommentsDetails()).then(() => {
      const actions = store.getActions();
      expect(actions[0]).toEqual({ type: COMMENTS_LIST_REQUEST });
    });
  });
});
