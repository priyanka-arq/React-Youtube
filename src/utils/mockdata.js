import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);

export const store = mockStore({
  auth: {
    accessToken:
      "ya29.A0ARrdaM9KG5nxzuU4bOhVbHKu2EDj2lKG36NY7wtm8IVsKLEbo3_O699EDk8IxvhccuIHPUyhQTT5pXxmCAavDL4iTI7iblxFuIXDONABwlhi8LQHbhvxGfMn27FnCksi2qJhg4hn7-IzIcHxildJ0RIn2ragrw",
    user: {
      name: "Priyanka Patel",
      photoURL:
        "https://lh3.googleusercontent.com/a-/AOh14GgEX3SB8FVF-3OLqmdT3zAW0vRL6alhlseVN0bn=s96-c",
    },
    error: null,
    loading: false,
  },
  commentList: [
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
    {
      kind: "youtube#commentThread",
      etag: "9SadOUrGkPQSt0sorsFijKTBL80",
      id: "Ugwybolefx_P4E6phah4AaABAg",
      snippet: {},
    },
  ],
  homeVideos: {
    videos: [
      {
        kind: "youtube#video",
        etag: "8XDUSTuq-9vHEgBH-g3R1QOmD30",
        id: "ATElufr0OiE",
        snippet: {
          publishedAt: "2022-01-31T16:27:42Z",
          channelId: "UCTNtRdBAiZtHP9w7JinzfUg",
          title:
            "Mudhal Nee Mudivum Nee - Title Track Video | Darbuka Siva | Sid Sriram | Thamarai",
          description:
            "A lilting melody from the famed #DarbukaSiva , #SidSriram and #Thamarai trio! Presenting the loop-worthy #MudhalNeeMudivumNee title track from the composer's directorial debut!\n\nHit play and enjoy the official Tamil song video of #MNMN now!\n\nStream now - https://SMI.lnk.to/MNMNTitleTrackDC\n\nComposer's Note:\n\nThe song Mudhal Nee Mudivum Nee for me is a collage of sounds from various musical soundscapes that relate to nostalgia and the wonderful memories associated with it. \n\nThe Sarodâ€™s classical texture converses with the voice to create that feeling, coupled with strings that form a sonic movement which often plays around with trancy layers of interesting shapes and patterns of sounds.\n\nDarbuka Siva\n\nMovie - Mudhal Nee Mudivum Nee\nSong - Mudhal Nee Mudivum Nee\nSinger - Sid Sriram, Darbuka Siva\nComposer - Darbuka Siva\nLyrics - Thamarai\nWritten and Directed by Darbuka Siva\nDOP - Sujith Saarang\nEditing - Sreejith Saarang\nProduced by Super Talkies - Sameer Bharat Ram\n\nMusicians:\nSarod: Debasmita Bhattacharya\nElectric Guitar, Acoustic guitar, Banjo, Ukulele: Keba Jeremiah\nKeyboard: Shyam Benjamin \nMusic arranged and programmed by Darbuka Siva\nMixed and Mastered by Balu Thankachan at 20dB Soundstudios, Chennai.\nAssisted by Elwin Joseph, Hariharan\n \nRecorded at \nSeed Studios, Chennai\n20 dB Soundstudios, Chennai\nRJ Studio, Kolkata\nRecording Engineers: Dheeleben, Hafeez, Avinash Sathish, Rana Mondal\n \nLyric Video Credits: \nArtwork Hand drawn by Purva Raghunath\nVideo Effects: Pradeep\nVideo Editing: Darbuka Siva\n\nMusic Label - Sony Music Entertainment India Pvt. Ltd.\n\nÂ© 2022 Sony Music Entertainment India Pvt. Ltd.\n\nSubscribe Now: http://bit.ly/SonyMusicSouthVevo\nSubscribe Now: http://bit.ly/SonyMusicSouthYT\nFollow us: https://www.instagram.com/sonymusic_south/\nFollow us: Twitter: https://twitter.com/SonyMusicSouth\nLike us: Facebook: https://www.facebook.com/SonyMusicSouth\n\nhttp://vevo.ly/TSdD3a",
          thumbnails: {
            default: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/default.jpg",
              width: 120,
              height: 90,
            },
            medium: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/mqdefault.jpg",
              width: 320,
              height: 180,
            },
            high: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/hqdefault.jpg",
              width: 480,
              height: 360,
            },
            standard: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/sddefault.jpg",
              width: 640,
              height: 480,
            },
            maxres: {
              url: "https://i.ytimg.com/vi/ATElufr0OiE/maxresdefault.jpg",
              width: 1280,
              height: 720,
            },
          },
          channelTitle: "SonyMusicSouthVEVO",
          tags: [
            "Sony Music South",
            "Sony Music",
            "Latest Song",
            "Tamil song",
            "Tamil Latest Song",
            "Tamil Latest Songs 2018",
            "Tamil Latest Songs 2019",
            "Latest songs in Tamil",
            "Songs in Tamil",
            "Latest Tamil Movies",
            "Tamil Movie Songs",
            "Latest Video Songs Tamil",
            "latest Tamil Love Songs",
            "Latest Tamil Hit Songs",
            "darbuka siva songs",
            "sid sriram songs",
            "sid sriram tamil hits",
            "darbuka siva tamil songs",
            "mudhalum nee mudivum nee trailer",
            "Darbuka Siva",
            "Mudhal Nee Mudivum Nee",
            "Sony",
            "Pvt.",
          ],
          categoryId: "10",
          liveBroadcastContent: "none",
          localized: {
            title:
              "Mudhal Nee Mudivum Nee - Title Track Video | Darbuka Siva | Sid Sriram | Thamarai",
            description:
              "A lilting melody from the famed #DarbukaSiva , #SidSriram and #Thamarai trio! Presenting the loop-worthy #MudhalNeeMudivumNee title track from the composer's directorial debut!\n\nHit play and enjoy the official Tamil song video of #MNMN now!\n\nStream now - https://SMI.lnk.to/MNMNTitleTrackDC\n\nComposer's Note:\n\nThe song Mudhal Nee Mudivum Nee for me is a collage of sounds from various musical soundscapes that relate to nostalgia and the wonderful memories associated with it. \n\nThe Sarodâ€™s classical texture converses with the voice to create that feeling, coupled with strings that form a sonic movement which often plays around with trancy layers of interesting shapes and patterns of sounds.\n\nDarbuka Siva\n\nMovie - Mudhal Nee Mudivum Nee\nSong - Mudhal Nee Mudivum Nee\nSinger - Sid Sriram, Darbuka Siva\nComposer - Darbuka Siva\nLyrics - Thamarai\nWritten and Directed by Darbuka Siva\nDOP - Sujith Saarang\nEditing - Sreejith Saarang\nProduced by Super Talkies - Sameer Bharat Ram\n\nMusicians:\nSarod: Debasmita Bhattacharya\nElectric Guitar, Acoustic guitar, Banjo, Ukulele: Keba Jeremiah\nKeyboard: Shyam Benjamin \nMusic arranged and programmed by Darbuka Siva\nMixed and Mastered by Balu Thankachan at 20dB Soundstudios, Chennai.\nAssisted by Elwin Joseph, Hariharan\n \nRecorded at \nSeed Studios, Chennai\n20 dB Soundstudios, Chennai\nRJ Studio, Kolkata\nRecording Engineers: Dheeleben, Hafeez, Avinash Sathish, Rana Mondal\n \nLyric Video Credits: \nArtwork Hand drawn by Purva Raghunath\nVideo Effects: Pradeep\nVideo Editing: Darbuka Siva\n\nMusic Label - Sony Music Entertainment India Pvt. Ltd.\n\nÂ© 2022 Sony Music Entertainment India Pvt. Ltd.\n\nSubscribe Now: http://bit.ly/SonyMusicSouthVevo\nSubscribe Now: http://bit.ly/SonyMusicSouthYT\nFollow us: https://www.instagram.com/sonymusic_south/\nFollow us: Twitter: https://twitter.com/SonyMusicSouth\nLike us: Facebook: https://www.facebook.com/SonyMusicSouth\n\nhttp://vevo.ly/TSdD3a",
          },
          defaultAudioLanguage: "en",
        },
        contentDetails: {
          duration: "PT5M36S",
          dimension: "2d",
          definition: "hd",
          caption: "false",
          licensedContent: true,
          regionRestriction: {
            blocked: ["US"],
          },
          contentRating: {},
          projection: "rectangular",
        },
        statistics: {
          viewCount: "2088681",
          likeCount: "133120",
          favoriteCount: "0",
          commentCount: "2850",
        },
      },
      {
        kind: "youtube#video",
        etag: "PK0np5HEbZ3Cz6FX3hZ7aSXu4As",
        id: "67Fuq_oRPMw",
        snippet: {
          publishedAt: "2022-01-31T03:42:51Z",
          channelId: "UC1J-EgBVeo-TUaC5m0t6bKg",
          title: "balloon and syringe experiment ðŸ¤© gone wrong ðŸ¤¯ #shorts",
          description: "",
          thumbnails: {
            default: {
              url: "https://i.ytimg.com/vi/67Fuq_oRPMw/default.jpg",
              width: 120,
              height: 90,
            },
            medium: {
              url: "https://i.ytimg.com/vi/67Fuq_oRPMw/mqdefault.jpg",
              width: 320,
              height: 180,
            },
            high: {
              url: "https://i.ytimg.com/vi/67Fuq_oRPMw/hqdefault.jpg",
              width: 480,
              height: 360,
            },
          },
          channelTitle: "Mr. Easy Tamil",
          tags: [
            "craft tamil channel",
            "craft tamil",
            "experiment in tamil",
            "experiment tamil",
            "easy experiment",
            "science experiment",
            "magic tricks",
            "balloon experiment gone wrong",
            "Mr easy tamil",
            "laminar flow experiment",
            "satisfying experiment",
          ],
          categoryId: "24",
          liveBroadcastContent: "none",
          defaultLanguage: "en-IN",
          localized: {
            title: "balloon and syringe experiment ðŸ¤© gone wrong ðŸ¤¯ #shorts",
            description: "",
          },
          defaultAudioLanguage: "ta",
        },
        contentDetails: {
          duration: "PT1M",
          dimension: "2d",
          definition: "hd",
          caption: "false",
          licensedContent: true,
          contentRating: {},
          projection: "rectangular",
        },
        statistics: {
          viewCount: "2531259",
          likeCount: "158897",
          favoriteCount: "0",
          commentCount: "66",
        },
      },
    ],
    activeCategory: "react",
    loading: false,
  },
  selectedVideos: { loading: false, videos: {}, error: null },
  channelDetails: { loading: false, channel: {}, error: null },
  subscriptionStatus: { subscriptionStatus: false },
  relatedVideos: { loading: false, videos: [], error: null },
  searchVideos: { loading: false, videos: [], error: null },
  subscriptionChannelVideos: { loading: false, videos: [], error: null },
  channelVideos: { loading: false, videos: [], error: null },
  likedVideos: { loading: false, videos: [], error: null },
});
