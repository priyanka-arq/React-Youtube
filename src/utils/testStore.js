import { createStore, applyMiddleware, combineReducers } from "redux";
import { rootReducer, middlewares } from "../redux/store";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

export const findByTestAtrr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};
export const testStore = (initialState = {}) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(
    createStore
  );

  return createStoreWithMiddleware(rootReducer, initialState);
};
// const mockStore = configureStore([thunk]);

// export const testStore = (state = {}) => {
//   return mockStore({
//     ...state,
//   });
// };
