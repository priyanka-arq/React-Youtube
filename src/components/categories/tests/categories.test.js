import Categories from "../Categories";
import React from "react";
import { Provider } from "react-redux";
import { mount, shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import toJSON from "enzyme-to-json";
import { findByTestAtrr } from "../../../utils/testStore";

let wrapper;
let props;
let handleClickSpy;
let handleClick;

const mockStore = configureMockStore([thunk]);

beforeEach(() => {
  props = {
    handleClickSpy: jest.fn(),
  };
  const store = mockStore({
    homeVideos: {},
    selectedVideo: {},
    channelDetails: {},
    subscriptionStatus: {},
    commentList: {},
    relatedVideos: {},
    searchedVideos: {},
    subscriptionChannelVideos: {},
    channelVideos: {},
    likedVideos: {},
  });

  wrapper = mount(
    <Provider store={store}>
      <Categories handleClick />
    </Provider>
  );
});

afterEach(() => {
  wrapper.unmount();
});

describe("Categories component", () => {
  describe("Should render categories  correctly ", () => {
    test("Should render categories  correctly ", () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it("should render a Categories component", () => {
      expect(wrapper.find("Categories").length).toEqual(1);
    });

    it("Should render without errors", () => {
      const component = findByTestAtrr(wrapper, "categoriesComponent");
      expect(component.length).toBe(1);
    });
  });
});

//##############
describe("Render span components ", () => {
  handleClick = jest.fn();
  const clickEventMocked = {
    getPopularVideos: jest.fn(),
    getVideosByCategory: jest.fn(),
  };
  it("should render a span component", () => {
    const component = findByTestAtrr(wrapper, "buttonComponent");
    expect(component.length).toBe(26);
  });

  it("Test onclick event ", () => {
    let active;
    active = wrapper.find("Categories").find(".active");
    // const component = findByTestAtrr(wrapper, "buttonComponent");
    // component.at(0).simulate("click", clickEventMocked);
    active.simulate("click", clickEventMocked);
    expect(active).toHaveLength(1);
    active = wrapper.find("Categories").find(".active");
    //const callBack = handleClick.mock.calls.length;

    //expect(clickEventMocked.getPopularVideos).toBeCalledTimes(1);
    //expect(clickEventMocked.getVideosByCategory).toBeCalledTimes(1);
  });
});
