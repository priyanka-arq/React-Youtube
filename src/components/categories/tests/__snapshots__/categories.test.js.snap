// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component Should render categories  correctly  Should render categories  correctly  1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Categories
    handleClick={true}
  >
    <div
      className="categories"
      data-test="categoriesComponent"
    >
      <HiChevronLeft>
        <IconBase
          attr={
            Object {
              "fill": "currentColor",
              "viewBox": "0 0 20 20",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z"
              fillRule="evenodd"
              key="0"
            />
          </svg>
        </IconBase>
      </HiChevronLeft>
      <span
        className="active"
        data-test="buttonComponent"
        key="0"
        onClick={[Function]}
      >
        All
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="1"
        onClick={[Function]}
      >
        Javascript
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="2"
        onClick={[Function]}
      >
        React js
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="3"
        onClick={[Function]}
      >
        Angular js
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="4"
        onClick={[Function]}
      >
        React Native
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="5"
        onClick={[Function]}
      >
        use of API
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="6"
        onClick={[Function]}
      >
        Redux
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="7"
        onClick={[Function]}
      >
        Poor Coder
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="8"
        onClick={[Function]}
      >
        Algorithm Art 
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="9"
        onClick={[Function]}
      >
        Meditation music
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="10"
        onClick={[Function]}
      >
        Music
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="11"
        onClick={[Function]}
      >
        Thrillers
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="12"
        onClick={[Function]}
      >
        Courses
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="13"
        onClick={[Function]}
      >
        Indian soap operas
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="14"
        onClick={[Function]}
      >
        ios
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="15"
        onClick={[Function]}
      >
        Conversation
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="16"
        onClick={[Function]}
      >
        Comedy
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="17"
        onClick={[Function]}
      >
        Live
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="18"
        onClick={[Function]}
      >
        Nature
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="19"
        onClick={[Function]}
      >
        T-Series
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="20"
        onClick={[Function]}
      >
        Music
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="21"
        onClick={[Function]}
      >
        Guitar
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="22"
        onClick={[Function]}
      >
        Bengali Songs
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="23"
        onClick={[Function]}
      >
        Coding
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="24"
        onClick={[Function]}
      >
        Cricket
      </span>
      <span
        className=""
        data-test="buttonComponent"
        key="25"
        onClick={[Function]}
      >
        Football
      </span>
      <HiChevronRight>
        <IconBase
          attr={
            Object {
              "fill": "currentColor",
              "viewBox": "0 0 20 20",
            }
          }
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
              fillRule="evenodd"
              key="0"
            />
          </svg>
        </IconBase>
      </HiChevronRight>
    </div>
  </Categories>
</Provider>
`;
